src/Passes/.clang-tidy

```diff
@@ -0,0 +1,103 @@
+readability-braces-around-statements,\
+cert-dcl*,\
+cert-env*,\
+cert-err52-cpp,\
+cert-err60-cpp,\
+cert-flp30-c,\
+clang-analyzer-security.FloatLoopCounter,\
+google-build-explicit-make-pair,\
+google-build-namespaces,\
+google-explicit-constructor,\
+google-readability-*,\
+google-runtime-operator,\
+hicpp-exception-baseclass,\
+hicpp-explicit-conversions,\
+hicpp-use-*,\
+modernize-avoid-bind,\
+modernize-loop-convert,\
+modernize-make-shared,\
+modernize-make-unique,\
+modernize-redundant-void-arg,\
+modernize-replace-random-shuffle,\
+modernize-shrink-to-fit,\
+modernize-use-bool-literals,\
+modernize-use-default-member-init,\
+modernize-use-emplace,\
+modernize-use-equals-default,\
+modernize-use-equals-delete,\
+modernize-use-noexcept,\
+modernize-use-nullptr,\
+modernize-use-override,\
+modernize-use-transparent-functors,\
+misc-*,\
+-misc-misplaced-widening-cast,\
+performance-*"
+
+WarningsAsErrors: '*'
+HeaderFilterRegex: '.*'
+
+CheckOptions:
+  # Configuration documentation: https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html
+  # Namespaces
+  - key:    readability-identifier-naming.NamespaceCase
+    value:  'lower_case'
+
+  # Classes and structs
+  - key:    readability-identifier-naming.AbstractClassPrefix
+    value:  'I'
+  - key:    readability-identifier-naming.ClassCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.StructCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.UnionCase
+    value:  'CamelCase'
+
+  # Class members
+  - key:    readability-identifier-naming.PrivateMemberCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.PrivateMemberSuffix
+    value:  '_'    
+  - key:    readability-identifier-naming.ProtectedMemberCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.ProtectedMemberSuffix
+    value:  '_'        
+
+  # Alias
+  - key:    readability-identifier-naming.TypeAliasCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.TypedefCase
+    value:  'CamelCase'
+
+  # Functions
+  - key:    readability-identifier-naming.FunctionCase
+    value:  'camelBack'
+  - key: readability-identifier-naming.IgnoreMainLikeFunctions
+    value: true
+
+  # Variables and parameters
+  - key:    readability-identifier-naming.VariableCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.LocalVariableCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.ParameterCase
+    value:  'lower_case'
+
+  # Globals, consts and enums
+  - key:    readability-identifier-naming.GlobalConstantCase
+    value:  'UPPER_CASE'
+  - key:    readability-identifier-naming.GlobalConstantPrefix
+    value:  'G_'
+  - key:    readability-identifier-naming.ConstantCase
+    value: 'UPPER_CASE'
+  - key:    readability-identifier-naming.EnumCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.EnumConstantCase
+    value:  'CamelCase'
```

Comment:
Thank you for providing the [link](https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html)!
It encouraged me to take a look at what's in there.

(Below is is an insane "pedanticness", feel free to ignore :-D)
After which I asked myself: "Do `EnumCase` and `EnumConstantCase` talk about the _name of the variable_ of the enumerated type or about the _name of the enumerated type_ itself?" The examples in the link (like `enum pre_foo_post { One, Two, Three }`) told me that it is the _data type itself_ (not the var name). That is why I'd suggest to move the `EnumCase` and `EnumConstantCase` to the `# Alias` section (and probably rename the section to `# Types and Type Aliases` or similar).

Reply:
I am happy to incorporate these changes. I will do this on the new branch I am working if you are happy with that?

Just a side note: Enums are a bit of a funny kind of entity - I see your point and why you'd want them together with Types and Type Aliases. The reason why they are with consts is that anonymous enums, i.e.
```
enum {
  One, 
  Two,
  Three
}
```
gives you enumerated constants which are not typed as a separated class.

Reply:
I've updated this in #1098.